### PR TITLE
Optimized Delaunay2D and made it more readable

### PR DESCRIPTION
**Pull request by D-T-666**
Execution takes about 35% of the time it took before

### CODE REVIEW

File: core/math/delaunay_2d.h [hunk 1]
```diff
@@ -48,50 +49,50 @@ class Delaunay2D {
...
 
-		real_t ab = p1.x * p1.x + p1.y * p1.y;
-		real_t cd = p2.x * p2.x + p2.y * p2.y;
-		real_t ef = p3.x * p3.x + p3.y * p3.y;
+		Vector2 O = (b * a.length_squared() - a * b.length_squared()) / (a.cross(b) * 2.0f);
 
-		Vector2 circum(
-				(ab * (p3.y - p2.y) + cd * (p1.y - p3.y) + ef * (p2.y - p1.y)) / (p1.x * (p3.y - p2.y) + p2.x * (p1.y - p3.y) + p3.x * (p2.y - p1.y)),
-				(ab * (p3.x - p2.x) + cd * (p1.x - p3.x) + ef * (p2.x - p1.x)) / (p1.y * (p3.x - p2.x) + p2.y * (p1.x - p3.x) + p3.y * (p2.x - p1.x)));
+		// Rotating by 90°
```
Comment (fire) [inline core/math/delaunay_2d.h:73]: Godot Engine style guides uses Sentence Case with punctuation.

Comment (D-T-666) [reply core/math/delaunay_2d.h]: fixed ✅

File: core/math/delaunay_2d.h [hunk 2]
```diff
@@ -48,50 +49,50 @@ class Delaunay2D {
...
 
-		return false;
+		return result;
 	}
 
 	static Vector<Triangle> triangulate(const Vector<Vector2> &p_points) {
 		Vector<Vector2> points = p_points;
 		Vector<Triangle> triangles;
 
+		// Get the bounding rect.
```
Comment (MewPurPur) [inline core/math/delaunay_2d.h:95]: We can check if the point count is 0-2 initially and return the empty triangles vector right away. Then we'll know there's a "first" point and we can initialize the rect right away, no need for the `if (i == 0)` condition.

But it probably won't amount to anything substantial other than in the 0-2 points case.

**Review by MewPurPur — APPROVED** [review]
A great simple implementation of 2D Delaunay! The code looks good and I measured roughly the same performance boost.

I'll note that the method is now going to return the triangles in a different order and the triangle vertices are going to be sorted like integers, but no code should be relying on any specific order for this particular algorithm.

**Pull request closed without merging**